### ISSUE TITLE
feat: Websocket support for status APIs

### ISSUE THREAD

**Issue by krzkaczor** [triggerdotdev/trigger.dev#828]
### Is your feature request related to a problem? Please describe.

I am surprised that old-school HTTP pooling is used in [react](https://github.com/triggerdotdev/trigger.dev/blob/main/packages/react/src/fetch.ts) package instead of something like websockets. 

### Describe the solution you'd like to see

Websockets could provide faster updates (but ofc there is some architecture overhead).  Is this something that is on your roadmap?

### Describe alternate solutions

-

**Comment by matt-aitken** [comment]
Yep we’ve talked about adding web socket and Server Sent Event support. We’d use Redis PubSub to make it easier to build I think.

@krzkaczor out of interest, which of the hooks are you using/thinking of using? We have a few different ones and I’m interested in figuring out the most valuable.

**Comment by krzkaczor** [comment]
I want to be able to track the progress of a job so I added a custom status to the job as defined in [docs](https://trigger.dev/docs/documentation/guides/react-hooks-statuses). On the client side, I use `useEventRunStatuses` hook (inside next.js app). 

For now, it's nothing more than a PoC but I am absolutely loving trigger.dev so far. I can't wait to build something production-ready with it  ;)

**Comment by matt-aitken** [comment]
Thanks!

I'm glad you said the Status hooks because we're going to be taking that system and turning it into a more powerful notification system. React hooks, web sockets, SSE etc will be possible ways to subscribe to it.
